packages/financial-templates-lib/src/price-feed/CryptoWatchPriceFeed.ts

```diff
@@ -84,9 +89,18 @@ export class CryptoWatchPriceFeed extends PriceFeedInterface {
       throw new Error(`${this.uuid}: undefined lastUpdateTime`);
     }
 
+    let twapLength = this.twapLength;
+    // Ancillary data might contain parameters that affect how we compute the historical price, such as the twapLength.
+    try {
+      const parsedAncillaryData = (parseAncillaryData(ancillaryData) as unknown) as SimplePriceAncillaryData;
+      twapLength = Number(parsedAncillaryData.twapLength);
+    } catch (err) {
+      // Could not parse ancillary data into utf8 dictionary, use default parameters.
+    }
```

Comment:
Are we sure we want to silently use default params if the ancillary data can't be parsed? Maybe we should enumerate the cases:
- Ancillary data is empty. Seems fine to use default params in this case. Not sure if parseAncillaryData errors in this case, but if not, this would end up looking identical to the case below.
- Ancillary data doesn't contain the twapLength field.  Seems fine to use default params in this case.
- Ancillary data can't be parsed to UTF8. Seems like this should throw, right?
- Ancillary data has a value for twapLength that can't be parsed into a number. Seems like this should also throw, right?

Thoughts?

Reply:
I agree with all these cases, and its pretty easy to implement actually: [commit](https://github.com/UMAprotocol/protocol/pull/3659/commits/7e257ee0f110ede8b1cd99b45150032d6c956a42)